src/store.c

```diff
@@ -351,14 +352,18 @@ int pthreads_store_write(zval *object, zval *key, zval *write) {
 				normal refcounting, we'll just never read the reference
 			*/
 			rebuild_object_properties(&threaded->std);
+			
+			pthreads_store_sync(object);
```

Comment:
I have again tested the script, without the pthreads_store_sync call. I must admit the memory isn't growing. My fault, I must have done something wrong.